viaq-jenkins/fields.yml

```diff
@@ -0,0 +1,493 @@
+        Data in this subgroup is stored for troublehsooting purposes mostly.
+      name: pipeline_metadata
+      fields:
+      - name: collector
+        type: group
+        description: >
+          The section contains metadata specific to the collector.
+        fields:
+        - name: hostname
+          type: string
+          index: not_analyzed
+          description: >
+            FQDN of the collector. It might be different from the FQDN of the actual emitter of the logs.
+
+        - name: ipaddr4
+          type: ip
+          description: >
+            IP address v4 of the collector server, can be an array.
+          fields:
+            - name: raw
+              ignore_above: 256
+              type: string
+              index: not_analyzed
+          norms:
+            enabled: false
+
+        - name: inputname
+          type: string
+          index: not_analyzed
+          description: >
+            how the log message was received on the collector whether it was TCP/UDP, or maybe imjournal/imfile.
+
+        - name: received_at
+          type: date
+          format: yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ||yyyy-MM-dd'T'HH:mm:ssZ||dateOptionalTime
+          description: >
+            Time when the message was received at the collector.
+
+        - name: name
+          type: string
+          example: shipshift
+          index: not_analyzed
+          description: >
+            The name of the collector
+
+      - name: normalizer
+        type: group
+        description: >
+          The section contains metadata specific to the normalizer.
+        fields:
+        - name: hostname
+          type: string
+          index: not_analyzed
+          description: >
+            FQDN of the normalizer.
+
+        - name: ipaddr4
+          type: ip
+          description: >
+            IP address v4 of the normalizer server, can be an array.
+          fields:
+            - name: raw
+              ignore_above: 256
+              type: string
+              index: not_analyzed
+          norms:
+            enabled: false
+
+        - name: inputname
+          type: string
+          index: not_analyzed
+          description: >
+            how the log message was received on the normalizer whether it was TCP/UDP.
+
+        - name: received_at
+          type: date
+          format: yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ||yyyy-MM-dd'T'HH:mm:ssZ||dateOptionalTime
+          description: >
+            Time when the message was received at the normalizer.
+
+        - name: name
+          type: string
+          example: rhci-logstash
+          index: not_analyzed
+          description: >
+            The name of the normalizer
+
+      - name: "@version"
+        index: not_analyzed
+        type: string
+        example: TODO
+        description: >
+          Version of “com.redhat.viaq” mapping the document is intended to adhere by the normalizer. It must be set by the normalizer. 
```

Comment:
pointer to see the generated asciidoc in readme for particular fields and how they're supposed to be used.